numpy/_typing/_array_like.py

```diff
@@ -21,6 +21,7 @@
     str_,
     bytes_,
 )
+from numpy._core.multiarray import StringDType
```

Comment:
I'm sure about the runtime, but at least in the stubs, `StringDType` is defined in `numpy.dtypes`. So if it's the same at runtime, and `numpy._core.multiarray` re-exports it, then it could be cleaner to import directly it from `np.dtypes`

Reply:
Agreed, for some reason the `np.dtypes` import breaks under reloading or lazy-loading. I'm not sure why it happens but it's a pre-existing issue that until now hasn't been problematic. I'd need to poke around in a C debugger to understand why it's happening, this was just a quick fix for the tests.